### PR TITLE
Read xns11 files

### PR DESCRIPTION
**Pull request by rhaDHI**

### CODE REVIEW

**Comment by rhaDHI** [comment]
@ecomodeller Do you run Black formatting manually or how does that work? I haven't tried it before but stumbled upon it in the guidelines.

**Comment by ecomodeller** [comment]
I use [auto-formatting in VS Code](https://code.visualstudio.com/docs/python/editing#_general-formatting-settings) and set the formatter to black. But that's not very important, I just  want to have reasonably formatted code without having to follow a long style guide document.

File: tests/test_xns11.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+from mikeio.xns11 import (
+    read, Xns11, QueryData, FileNotOpenedError, DataNotFoundInFile
+)
+
+
+def test_query_validate():
+    # Bad topo id type
+    with pytest.raises(TypeError):
+        QueryData(666)
```
Comment (ecomodeller) [inline tests/test_xns11.py:11]: Is it necessary to raise an exception? Isn't it possible to convert the supplied integer (in this case) to a string for a better user experience?

File: mikeio/xns11.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        self.__topoids = None
+        # Load the file on initialization
+        self._load_file()
+
+    def _load_file(self):
+        """Load the file."""
+        if not os.path.exists(self.file_path):
+            raise FileExistsError(f"File {self.file_path} does not exist.")
+        self.file = CrossSectionDataFactory()
+        self.file = self.file.Open(
```
[the remark this answers is not in the document]
Comment (mmfontana) [reply mikeio/xns11.py:100]: There is no need. Thanks, @ecomodeller. 

```suggestion
        self.file = CrossSectionDataFactory.self.file.Open(
```

[the remark this answers is not in the document]
Comment (mmfontana) [reply mikeio/xns11.py:100]: There is no need. Thanks, @ecomodeller.
```suggestion
        self.file = CrossSectionDataFactory(.Open(
```

[the remark this answers is not in the document]
Comment (mmfontana) [reply mikeio/xns11.py:100]: There is no need. Thanks, @ecomodeller.
```suggestion
        self.file = CrossSectionDataFactory().Open(
```

File: mikeio/xns11.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+            location.Chainage = chainage.value
+            geometry = (self.file.FindClosestCrossSection(
+                location, topoid.value)
+                .BaseCrossSection.Points)
+            x, z = [], []
+            for i in range(geometry.Count):
+                x.append(geometry.LstPoints[i].X)
+                z.append(geometry.LstPoints[i].Z)
+            x_name = f"x {topoid.value} {reach.value} {str(chainage.value)}"
+            z_name = f"z {topoid.value} {reach.value} {str(chainage.value)}"
```
[the remark this answers is not in the document]
Comment (mmfontana) [reply mikeio/xns11.py:173]: Absolutely not.  Thanks, @ecomodeller.
```suggestion
            x_name = f"x {topoid.value} {reach.value} {chainage.value}"
            z_name = f"z {topoid.value} {reach.value} {chainage.value}"
```

File: mikeio/xns11.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmfontana) [reply mikeio/xns11.py:100]: There is no need. Thanks, @ecomodeller.
```suggestion
        self.file = CrossSectionDataFactory().Open(
```

File: tests/test_xns11.py (hunk 1 above)
Comment (mmfontana) [reply tests/test_xns11.py:11]: I am not sure if using numeric values for topo-id names is a good practice (same for reach names).

**Comment by mmfontana** [comment]
Hello @ecomodeller and @rhaDHI,

I have noticed that the .xns11 branch is still open. Could I help with some left task to finally close the branch? Please let me know.

Thanks.

**Comment by rhaDHI** [comment]
@mmfontana We simply forgot to delete the branch after merging into master. If you would like to add more features, simply create your own branch based on this xns11 branch and PR back. Otherwise I will delete it. Is that what you mean?

**Comment by mmfontana** [comment]
@rhaDHI Sorry, I understand now. You can delete it. I will keep working on my forked. Then I will PR to DHI:master when I got something solid, is it fine? Thanks, it's a pleasure to work with you guys.

**Pull request merged**